source/agora/consensus/protocol/Nominator.d

```diff
@@ -750,21 +756,28 @@ extern(D):
         const Scalar block_challenge = block_hash;
         // Fetch the R from enrollment commitment for signing validator
         const CR = this.enroll_man.getCommitmentNonce(block_sig.public_key, block_sig.height);
+        // Get the preimage at ledger height
+        const preimage = this.enroll_man.getValidatorPreimageAt(block_sig.public_key, Height(block_sig.height - 1));
+        if (preimage == Hash.init)
+        {
+            log.warn("No preimage for {} to sign block at height {}", block_sig.public_key, block_sig.height);
+            return false;
+        }
```

Comment:
If we don't know the pre-image we shouldn't (couldn't) have externalized the block, because we used it for the random seed.

Reply:
But we call this also in `SCP confirm` which I think is before we externalize.

Reply:
The same reasoning applies: The process of creating the block should not have built a block which is not consistent.

Reply:
I left this check as this function is called when a block signature is received over the network. We want to ignore the signature if we do not have the `pre-image` otherwise the node could send us a block signature without revealing the `preimage` and cause us problems.